crates/threshold-bls/src/sig/sig.rs

```diff
@@ -65,28 +65,44 @@ pub trait SignatureScheme: Scheme {
     fn verify(public: &Self::Public, msg: &[u8], sig: &[u8]) -> Result<(), Self::Error>;
 }
 
-/// Extension trait over `SignatureScheme` which provides signing & verification methods
-/// which do not hash the message.
-pub trait SignatureSchemeExt: SignatureScheme {
-    /// Signs the message with the provided private key and returns a serialized signature. This
-    /// method **will not** hash the message before signing it. It should be used for
-    /// blind-signature related functionalities. In other cases, prefer `Signature`
-    fn sign_without_hashing(private: &Self::Private, msg: &[u8]) -> Result<Vec<u8>, Self::Error>;
-
-    /// Verifies that the signature on the provided message was produced by the public key.
-    /// This method **will not** h the message. It should be used when verifying blind signatures
-    /// by parties that do not have access to the blinding factor
-    fn verify_without_hashing(
-        public: &Self::Public,
-        msg: &[u8],
-        sig: &[u8],
-    ) -> Result<(), Self::Error>;
-}
-
-/// Blinder holds the functionality of blinding and unblinding a message. It is
-/// not to be used alone but in combination with a signature scheme or a
-/// threshold scheme.
-pub trait Blinder {
+/// BlindScheme is a signature scheme where the message can be blinded before
+/// signature so the signer does not know the real message. The signature can
+/// later be "unblinded" as to reveal a valid signature over the initial
+/// message.
+///
+/// ```
+///  # #[cfg(feature = "bls12_381")]
+///  # {
+///  use rand::prelude::*;
+///  use threshold_bls::{
+///     sig::{BlindScheme,SignatureScheme, Scheme, G2Scheme},
+///     group::{Element, Point}
+///  };
+///  use threshold_bls::curve::bls12381::PairingCurve as PC;
+///
+///  let msg = vec![1,9,6,9];
+///  let (private,public) = G2Scheme::<PC>::keypair(&mut thread_rng());
+///  // we first blind the message so the signers don't know the real underlying
+///  // message they are signing.
+///  let (token, blinded_msg) = G2Scheme::<PC>::blind_msg(&msg,&mut thread_rng());
+///  // this method is called on the signers, that sign blindly.
+///  let blinded_sig = G2Scheme::<PC>::blind_sign(&private,&blinded_msg).unwrap();
+///  // this method can be called by a third party that is able to verify if a
+///  // blinded signature is a a valid one even without having access to the
+///  // clear message.
+///  G2Scheme::<PC>::blind_verify(&public,&blinded_msg,&blinded_sig)
+///        .expect("blinded signatures should be correct");
+///  // the owner of the message can then unblind the signature to reveal a
```

Comment:
```suggestion

///  // the owner of the message can then unblind the signature to reveal a
```

Let's have spaces between each distinct step so that it's easier to read?